### PR TITLE
initial config, is missing fix the update on reload

### PR DESCRIPTION
**Pull request by 01001110J**
# Ticket
https://agile-jira.pearson.com/browse/PADV-1414

# Description
In this PR is fixed the error at the moment to open a new tab, the reference of the institution is lost.

## Changes made
- Refactor in the selector, is been moved to a different file, is included the respective test.
- Added new custom hook to handle the query param using the redux context.
- Refactor in ALL application links to reflect the institution selected. 
- Added new dep to help at the moment to test a custom hook.
- Refactor to the function resposable to format the select option along with it test.

## Screenshot
image

## How to test
- Make sure to have the devstack runnint
- Instal deps `@testing-library/react-hooks` has been added.
```bash
  npm i 
```
- Go to http://localhost:1980/
- The URL must change automatically to reflect the current institution id.
 -- If you have multiple institutions, change it using the selector, then copy the new url, you should be able to recover the institution selected in the previous tab, along with it information.

### CODE REVIEW

File: src/features/Main/InstitutionSelector/index.jsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+} from 'react';
+import { Row, Col } from '@edx/paragon';
+import { Select } from 'react-paragon-topaz';
+import { useLocation, useHistory } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+
+import { formatSelectOptions } from 'helpers';
+
+import { INSTITUTION_QUERY_ID } from 'features/constants';
+import { updateSelectedInstitution } from 'features/Main/data/slice';
```
Comment (AuraAlba) [inline src/features/Main/InstitutionSelector/index.jsx:13]:
```suggestion

import { Row, Col } from '@edx/paragon';
import { Select } from 'react-paragon-topaz';

import { useLocation, useHistory } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';

import { formatSelectOptions } from 'helpers';
import { INSTITUTION_QUERY_ID } from 'features/constants';

import { updateSelectedInstitution } from 'features/Main/data/slice';
```

File: src/features/Main/index.jsx
Comment (AuraAlba) [inline src/features/Main/index.jsx:1]: I don't see the institution param when I open the platform, neither if I open a new tab using one of the links in the tables

**Review by AuraAlba — CHANGES_REQUESTED** [review]

File: src/features/Main/index.jsx
Comment (01001110J) [reply src/features/Main/index.jsx:1]: What :face_exhaling:  could you add more details?

- Which table?
- Which url?
- This happens in the entire platform?
- This happen on slow connection?

File: src/features/Main/LinkWithQuery/index.jsx
Comment (AuraAlba) [inline src/features/Main/LinkWithQuery/index.jsx:1]: I lost the query in the view all button in dashboard

**Review by AuraAlba — CHANGES_REQUESTED** [review]

File: src/features/Dashboard/InstructorAssignSection/ClassCard.jsx [hunk 2]
```diff
@@ -5,14 +5,16 @@ import PropTypes from 'prop-types';
...
 
 import 'features/Dashboard/InstructorAssignSection/index.scss';
 
 const ClassCard = ({ data }) => {
   const history = useHistory();
+  const addQueryParam = useInstitutionIdQueryParam();
 
   const handleManageButton = () => {
-    history.push(`/manage-instructors/${encodeURIComponent(data?.masterCourseName)}/${encodeURIComponent(data?.className)}?classId=${data?.classId}&previous=dashboard`);
+    history.push(addQueryParam(`/manage-instructors/${encodeURIComponent(data?.masterCourseName)}/${encodeURIComponent(data?.className)}?classId=${data?.classId}&previous=dashboard`));
```
Comment (AuraAlba) [inline src/features/Dashboard/InstructorAssignSection/ClassCard.jsx:17]: Plese add the addqueryparam in all history.push of the platform

**Review by AuraAlba — CHANGES_REQUESTED** [review]

**Review by AuraAlba — APPROVED** [review]

**Review by anfbermudezme — APPROVED** [review]

**Pull request merged**